ratelimits/limit.go

```diff
@@ -244,6 +274,23 @@ func newLimitRegistry(defaults, overrides string) (*limitRegistry, error) {
 	return registry, nil
 }
 
+// newLimitRegistryWithData takes a map that is already populated with limit
+// data, instead of a filename from which to load limit data. Overrides are not
+// supported.
+func newLimitRegistryWithData(defaults limits) (*limitRegistry, error) {
```

Comment:
I think we've developed one too many layers of abstraction/indirection here. I _think_ we can take a slightly simpler approach -- let me know if this proposal doesn't make sense, or wouldn't work for some reason I'm missing.

* `loadAndParse[Default|Override]Limits` cease to exist -- only their underlying functions (i.e. load, and parse) exist.
* The `load` functions both return collections of `LimitConfig`, not `limit`, since that's all the data that's actually baked into those files. The `parse` functions convert `LimitConfig`s into `limit`s.
* `newLimitRegistryWithData` takes default and override collections of LimitConfigs, calls the `parse` functions on them, and produces a registry.
* `newLimitRegistry` takes default and override filenames, calls the `parse` functions on them, and then calls `newLimitRegistryWithData` to finish the job

So overall it would look very roughly (i.e. I'm ignoring error handling) like:
```go
func newLimitRegistry(defaults, overrides string) *limitRegistry {
  return newLimitRegistryWithData(loadDefaults(defaults), loadOverrides(overrides))
}

func newLimitRegistryWithData(defaults, overrides collection[LimitConfig]) *limitRegistry {
  return &limitRegistry{parseDefaults(defaults), parseOverrides(overrides)}
}
```

I think the difficulty of this approach is that it requires `newLimitRegistryWithData` to be able to take a collection of in-memory overrides. That's apparently somewhat messy, due to the map-of-maps structure of an overrides file, so you've side-stepped it in this PR. But I think it should be solvable without too much trouble? Is that where I'm wrong?

Reply:
You're right, I had side-stepped working with the overrides because of their more complex structure, which is solvable. There's also a fair amount of refactoring to do in `limit_test.go` if I change the `load` functions much more. Also very solvable; I'd just hoped to land a smaller change faster.

I'll work on this and see if I make good progress.